src/Catalyst.Core.Modules.Kvm/DeltaExecutor.cs

```diff
@@ -0,0 +1,600 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Linq;
+using Catalyst.Abstractions.Cryptography;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Modules.Cryptography.BulletProofs;
+using Catalyst.Protocol.Deltas;
+using Catalyst.Protocol.Transaction;
+using Google.Protobuf;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Extensions;
+using Nethermind.Core.Specs;
+using Nethermind.Dirichlet.Numerics;
+using Nethermind.Evm;
+using Nethermind.Evm.Precompiles;
+using Nethermind.Evm.Tracing;
+using Nethermind.Store;
+using Serilog;
+using Serilog.Events;
+
+namespace Catalyst.Core.Modules.Kvm
+{
+    /// <summary>
+    ///     Delta executor is responsible for executing delta entries code on the VM. Its responsibility is to prepare
+    ///     the execution environment and initial state of the VM as well as support some contract deployment operations.
+    ///     There is a minor responsibility issue in the way the code deposits are handled - depending on the execution
+    ///     level the deposit may happen inside the <see cref="VirtualMachine" /> or <see cref="DeltaExecutor" /> code.
+    /// </summary>
+    public class DeltaExecutor : IDeltaExecutor
+    {
+        private readonly ICryptoContext _cryptoContext = new FfiWrapper();
```

Comment:
CryptoContext can be injected `public DeltaExecutor(ICryptoContext cryptoContext,`

Reply:
at the time of writing it was a method used in other classes, I will change it

Reply:
Fixed, committed